build.gradle

```diff
@@ -736,123 +763,58 @@ allModuleNames.split(',').each{ moduleName ->
-
-                    // Souceset folders of the current module
-                    sourceSets[moduleName+'_gwt'].java.srcDirs,
-                    sourceSets[moduleName+'_gwt'].output.resourcesDir,
-                    sourceSets[moduleName+'_gwt'].output.classesDir
-                ]
-            }
-
-
-            args =
-                    [
-                        gwtModule,
-                        // Your GWT module
-                        '-war',
-                        buildDir,
-                        '-logLevel',
-                        'ERROR',
-                        '-localWorkers',
-                        '2',
-                        '-style',
-                        'obfuscated',
-                        '-extra',
-                        extraDir,
-                        '-strict'
-                        // '-draftCompile' // Speeds up compile with 25%
-                    ]
-            jvmArgs = [ '-Dgwt.jjs.permutationWorkerFactory=com.google.gwt.dev.ThreadedPermutationWorkerFactory']
-            
-
-            maxHeapSize = max_heap_size
+    if (gwtModule != null) {
+        task "gwt_$moduleName"(dependsOn: gwtClasses, type: org.opencms.GwtCompiler) {
+            opencmsModuleName = moduleName
+            gwtModuleName = gwtModule
+            draft = project.draft.toBoolean()
+            sourceSetMain sourceSets.main
+            sourceSetModules sourceSets.modules
+            sourceSetGwt sourceSets.gwt
+            maxHeapSize max_heap_size
         }
 
         tasks["jar_$moduleName"].dependsOn("gwt_$moduleName")
     }
 }
 
-tasks.findAll{ task -> task.name.startsWith('dist_')}.each{ dist_task ->
-    if (dist_task.requiresJar.toBoolean()){
-        dist_task.dependencies.each{ dep ->
-            def depCompileName = 'compile'+dep.replaceAll('\\.','')+'java'
-            tasks.findAll{ comp_task -> comp_task.name.toLowerCase().equals(depCompileName)}.each {comp_task ->
-                sourceSets[dist_task.moduleName].compileClasspath += files(sourceSets[dep].output.classesDir) { builtBy comp_task.name }
+tasks.findAll { task -> task.name.startsWith('dist_') }.each { dist_task ->
+    if (dist_task.requiresJar.toBoolean()) {
+        dist_task.dependencies.each { dep ->
+            def depCompileName = 'compile' + dep.replaceAll('\\.', '') + 'java'
+            tasks.findAll { comp_task -> comp_task.name.toLowerCase().equals(depCompileName) }.each { comp_task ->
+                sourceSets[dist_task.moduleName].compileClasspath += comp_task.outputs.files
                 // we need to manipulate the classpath of the java doc task specifically
-                tasks['javadoc_'+dist_task.moduleName].classpath += files(sourceSets[dep].output.classesDir) { builtBy comp_task.name }
+                tasks['javadoc_' + dist_task.moduleName].classpath += comp_task.outputs.files
             }
         }
-        tasks['jar_'+dist_task.moduleName].dependsOn{
```

Comment:
Unnecessary. From:

     task "jar_$moduleName" ... from sourceSets[moduleName].output

, gradle already discovers the dependency `jar_X` -> `XClasses` (and `XClasses` in turn, depend on `compileXJava` and `processXJava`